sdk/textanalytics/ai-text-analytics/test/public/utils/recordedClient.ts

```diff
@@ -4,71 +4,60 @@
 import { Context } from "mocha";
 
 import {
+  assertEnvironmentVariable,
   env,
   Recorder,
-  record,
-  RecorderEnvironmentSetup,
-  isLiveMode,
+  RecorderStartOptions,
 } from "@azure-tools/test-recorder";
-import { TokenCredential, ClientSecretCredential } from "@azure/identity";
-import { isNode, createXhrHttpClient } from "@azure/test-utils";
+import { TokenCredential } from "@azure/identity";
+import { createTestCredential } from "@azure-tools/test-credential";
 
 import { AzureKeyCredential, TextAnalyticsClient, TextAnalyticsClientOptions } from "../../../src/";
-import "./env";
 
-const replaceableVariables: { [k: string]: string } = {
+const envSetupForPlayback: { [k: string]: string } = {
   AZURE_CLIENT_ID: "azure_client_id",
   AZURE_CLIENT_SECRET: "azure_client_secret",
   AZURE_TENANT_ID: "88888888-8888-8888-8888-888888888888",
   TEXT_ANALYTICS_API_KEY: "api_key",
   // Second API key
   TEXT_ANALYTICS_API_KEY_ALT: "api_key_alt",
   ENDPOINT: "https://endpoint",
-  TEXT_ANALYTICS_RECOGNIZE_CUSTOM_ENTITIES_PROJECT_NAME: "project_name",
-  TEXT_ANALYTICS_RECOGNIZE_CUSTOM_ENTITIES_DEPLOYMENT_NAME: "deployment_name",
-  TEXT_ANALYTICS_SINGLE_CATEGORY_CLASSIFY_PROJECT_NAME: "project_name",
-  TEXT_ANALYTICS_SINGLE_CATEGORY_CLASSIFY_DEPLOYMENT_NAME: "deployment_name",
-  TEXT_ANALYTICS_MULTI_CATEGORY_CLASSIFY_PROJECT_NAME: "project_name",
-  TEXT_ANALYTICS_MULTI_CATEGORY_CLASSIFY_DEPLOYMENT_NAME: "deployment_name",
+  TEXT_ANALYTICS_RECOGNIZE_CUSTOM_ENTITIES_PROJECT_NAME: "sanitized",
+  TEXT_ANALYTICS_RECOGNIZE_CUSTOM_ENTITIES_DEPLOYMENT_NAME: "sanitized",
+  TEXT_ANALYTICS_SINGLE_CATEGORY_CLASSIFY_PROJECT_NAME: "sanitized",
+  TEXT_ANALYTICS_SINGLE_CATEGORY_CLASSIFY_DEPLOYMENT_NAME: "sanitized",
+  TEXT_ANALYTICS_MULTI_CATEGORY_CLASSIFY_PROJECT_NAME: "sanitized",
+  TEXT_ANALYTICS_MULTI_CATEGORY_CLASSIFY_DEPLOYMENT_NAME: "sanitized",
 };
 
-const environmentSetup: RecorderEnvironmentSetup = {
-  replaceableVariables,
-  customizationsOnRecordings: [
-    (recording: string): string =>
-      recording.replace(/"access_token"\s?:\s?"[^"]*"/g, `"access_token":"access_token"`),
-    // If we put ENDPOINT in replaceableVariables above, it will not capture
-    // the endpoint string used with nock, which will be expanded to
-    // https://<endpoint>:443/ and therefore will not match, so we have to do
-    // this instead.
-    (recording: string): string => {
-      const replaced = recording.replace("endpoint:443", "endpoint");
-      return replaced;
-    },
-  ],
-  queryParametersToSkip: [],
+const recorderStartOptions: RecorderStartOptions = {
+  envSetupForPlayback,
+  sanitizerOptions: {
+    generalSanitizers: [
+      {
+        regex: true,
+        target: `"access_token"\s?:\s?"[^"]*"`,
+        value: `"access_token":"access_token"`,
+      },
+    ],
```

Comment:
This "access_token" was meant for AAD I believe.
Shouldn't need if we are not recording AAD traffic.